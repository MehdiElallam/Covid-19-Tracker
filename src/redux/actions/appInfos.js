import { GET_PAGE_TITLE } from '../types'

export const changePageTitle = (title) => {
    
    return (dispatch) => {
        
        dispatch({
            type : GET_PAGE_TITLE,
            payload : title
        })
    }
    
}