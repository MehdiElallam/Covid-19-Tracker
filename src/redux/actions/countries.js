import {
    GET_WORLD_COUNTRIES_REQUEST, 
    GET_WORLD_COUNTRIES_SUCCESS, 
    GET_WORLD_COUNTRIES_ERROR,
} from '../types';
import {instance } from '../../utils/config'



export const fetchWorldCountriesRequest = () => {
    return {
        type : GET_WORLD_COUNTRIES_REQUEST
    }
}

export const fetchWorldCountriesSuccess = (countries) => {
    return {
        type : GET_WORLD_COUNTRIES_SUCCESS,
        payload : countries
    }
}

export const fetchWorldCountriesError = (error) => {
    return {
        type : GET_WORLD_COUNTRIES_ERROR,
        payload : error
    }
}

export const fetchWorldCountries = () => {
    
    return async (dispatch) => {
        
        dispatch(fetchWorldCountriesRequest)
        try {
            const countries = await instance.get('/countries')
            dispatch(fetchWorldCountriesSuccess(countries.data))
        }
        catch(error){
            const errorMsg = error.message
            dispatch(fetchWorldCountriesError(errorMsg))
        }
    }   
}