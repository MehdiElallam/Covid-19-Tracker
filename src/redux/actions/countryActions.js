import {GET_COUNTRIES_REQUEST, GET_COUNTRIES_SUCCESS, GET_COUNTRIES_ERROR} from '../types'
import axios from 'axios'
import { url } from '../../utils/config'



export const fetchCountriesRequest = () => {
    return {
        type : GET_COUNTRIES_REQUEST
    }
}

export const fetchCountriesSuccess = (countries) => {
    return {
        type : GET_COUNTRIES_SUCCESS,
        payload : countries
    }
}

export const fetchCountriesError = (error) => {
    return {
        type : GET_COUNTRIES_ERROR,
        payload : error
    }
}

export const fetchCountries = () => {
    
    return (dispatch) => {
        
        dispatch(fetchCountriesRequest)
        
        axios.get(`${url}/countries`)
        .then(response => {
            
            const countries = response.data
            dispatch(fetchCountriesSuccess(countries))
        })
        .catch(error => {
            
            const errorMsg = error.message
            dispatch(fetchCountriesError(errorMsg))
        })
    }
    
}