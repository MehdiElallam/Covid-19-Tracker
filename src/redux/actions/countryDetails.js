import {
    GET_COUNTRY_DATA_REQUEST, 
    GET_COUNTRY_DATA_SUCCESS, 
    GET_COUNTRY_DATA_ERROR,
    GET_COUNTRY_HISTORICAL_REQUEST,
    GET_COUNTRY_HISTORICAL_SUCCESS,
    GET_COUNTRY_HISTORICAL_ERROR

} from '../types'
import { instance } from '../../utils/config'

export const fetchCountryDataRequest = () => {
    return {
        type : GET_COUNTRY_DATA_REQUEST,
    }
}

export const fetchCountryDataSuccess = (data) => {
    return {
        type : GET_COUNTRY_DATA_SUCCESS,
        payload : data
    }
}

export const fetchCountryDataError = (error) => {
    return {
        type : GET_COUNTRY_DATA_ERROR,
        payload : error
    }
}

export const fetchCountryData = (country) => {
    
    return async(dispatch) => {
        
        dispatch(fetchCountryDataRequest)
        try {
            const countryDetails = await instance.get(`/countries/${country}`)
            dispatch(fetchCountryDataSuccess(countryDetails.data))
        }
        catch(error) {
            
            const errorMsg = error.message
            dispatch(fetchCountryDataError(errorMsg))
        }
    }
    
}

export const fetchCountryHistoricalRequest = () => {
    return {
        type : GET_COUNTRY_HISTORICAL_REQUEST,
    }
}

export const fetchCountryHistoricalSuccess = (data) => {
    return {
        type : GET_COUNTRY_HISTORICAL_SUCCESS,
        payload : data
    }
}

export const fetchCountryHistoricalError = (error) => {
    return {
        type : GET_COUNTRY_HISTORICAL_ERROR,
        payload : error
    }
}

export const fetchCountryHistorical = (country) => {
    
    return async(dispatch) => {
        
        dispatch(fetchCountryHistoricalRequest)
        try {
            const countryHistorical = await instance.get(`/historical/${country}?lastdays=365`)
            dispatch(fetchCountryHistoricalSuccess(countryHistorical.data))
        }
        catch(error) {
            
            const errorMsg = error.message
            dispatch(fetchCountryHistoricalError(errorMsg))
        }
    }
    
}