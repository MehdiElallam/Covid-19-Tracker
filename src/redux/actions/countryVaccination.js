import {
    GET_COUNTRY_VACCINATION_REQUEST, 
    GET_COUNTRY_VACCINATION_SUCCESS, 
    GET_COUNTRY_VACCINATION_ERROR,

} from '../types'
import { instance } from '../../utils/config'


export const fetchCountryVaccinationRequest = () => {
    return {
        type : GET_COUNTRY_VACCINATION_REQUEST,
    }
}

export const fetchCountryVaccinationSuccess = (data) => {
    return {
        type : GET_COUNTRY_VACCINATION_SUCCESS,
        payload : data
    }
}

export const fetchCountryVaccinationError = (error) => {
    return {
        type : GET_COUNTRY_VACCINATION_ERROR,
        payload : error
    }
}

export const fetchCountryVaccination = (country) => {
    
    return async(dispatch) => {
        
        dispatch(fetchCountryVaccinationRequest)
        try {
            const countryVaccination = await instance.get(`/vaccine/coverage/countries/${country}?lastdays=365`)
            dispatch(fetchCountryVaccinationSuccess(countryVaccination.data))
        }
        catch(error){
            const errorMsg = error
            dispatch(fetchCountryVaccinationError(errorMsg))
        }
    }
    
}