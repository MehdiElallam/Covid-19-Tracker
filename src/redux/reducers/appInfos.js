import { GET_PAGE_TITLE, DEFAULT_COUNTRY } from '../types'
 
const initialState = {
    infos : {
        title : 'Daily update',
        countryCode : 'UK'
    }
}

export default function appReducer (state = initialState, action){
    switch(action.type){
        case GET_PAGE_TITLE:
            return {
                ...state,
                infos : {
                    ...state.infos,
                    title : action.payload
                }
                
            }

        case DEFAULT_COUNTRY:
                return {
                    ...state,
                    infos : {
                        ...state.infos,
                        countryCode : action.payload
                    }
                    
                }
        
        default: return state
    }

}
