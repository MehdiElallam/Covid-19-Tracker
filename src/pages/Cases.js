import React, { useEffect, useState} from 'react'
import { useSelector, useDispatch } from 'react-redux'
import Container from 'react-bootstrap/Container'
import { Row, Col } from 'react-bootstrap'
import InfoBox from '../components/InfoBox'
import { changePageTitle } from '../redux/actions/appInfos'


export default function Cases() {
    
    const countryDetails = useSelector(state => state.countryDetails)
    const [dailyCases, setdailyCases] = useState(0)
    const [totalCases, setTotalCases] = useState(0)
    const [trendMonth, setTrendMonth] = useState(0)
    const dispatch = useDispatch();

    useEffect(() => {
        
        dispatch(changePageTitle('Cases'))
        !countryDetails.countryLoading && setdailyCases(countryDetails.country.total.todayCases)
        !countryDetails.countryLoading && setTotalCases(countryDetails.country.total.cases)
        !countryDetails.historicalLoading && setTrendMonth(Math.max(...Object.values(countryDetails.country.historical.timeline.cases)))

 
    }, [dispatch, countryDetails])
    
    
    return (
        <Container>
            <Row>
                <Col xs={6} md={4}>
                    <InfoBox title="Daily" number={dailyCases} />
                </Col>
                <Col xs={6} md={4}>
                    <InfoBox title="Total" number={totalCases} />
                </Col>
                <Col xs={6} md={4}>
                    <InfoBox title="Trend of the year" number={trendMonth} />
                </Col>
            </Row>
        </Container>
    )
}
