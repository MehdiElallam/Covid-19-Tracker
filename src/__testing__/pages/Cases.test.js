import React from 'react';
import { shallow } from 'enzyme';
import {Provider} from 'react-redux';
import Cases from '../../pages/Cases';
import { changePageTitle } from '../../redux/actions/appInfos'
import configureStore from 'redux-mock-store';

const mockStore = configureStore([]);

describe("Rendring Cases page", () => {
    let store;
    let component;

    beforeEach(() => {
        store = mockStore({
            country : {},
        });

        store.dispatch = jest.fn();

        component = shallow(
            <Provider store={store}>
                <Cases />
            </Provider>   
        );
    });

    it('should render with given state from mocked redux store', () => {
        expect(component).toMatchSnapshot();
    });

    it('should dispatch an action on Load', () => {
        store.dispatch(changePageTitle('Cases'))
        expect(store.dispatch).toHaveBeenCalledTimes(1);
    });

});