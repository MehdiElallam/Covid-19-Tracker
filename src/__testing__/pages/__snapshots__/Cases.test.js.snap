// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Rendring Cases page should render with given state from Redux store 1`] = `ShallowWrapper {}`;

exports[`Rendring Cases page should render with given state from mocked redux store 1`] = `ShallowWrapper {}`;
